### PR TITLE
Microbombs and Macrobombs, Nuke Ops start with Microbombs once more

### PR DESCRIPTION
**Pull request by Iamgoofball**
nhfq0oc
Changes up how Explosive Implants work.
They have become Microbomb Implants. They cost 1 TC each. Each implant you inject increases the size of the explosion. You gib when you activate them.
A Macrobomb implant was added for 20 TC, for those folks who want to buy 20 microbombs and inject them all. It is the equivalent of injecting 20 microbombs, aka 5, 10, 20, the maxcap.

Nuke ops start with microbomb implants again.

Supplementary video:
https://www.youtube.com/watch?v=s4VlruVG81w

### CODE REVIEW

**Comment by palpatine213** [comment]
:+1: Hell yes!

**Comment by Midaychi** [comment]
Neat in theory but sounds incredibly tedious in practice. Hope you like implanters.

**Comment by Lumipharon** [comment]
Neat, but still means paying 20tc to deny gear for the whole team. That's 10 spare mags of whatever, a shield/dual sword + change, etc etc.

Being able to bomb cap is pretty hilarious though.

Is the bomb core small or large? If it's small you could be MLGpr0 and implant it in yourself and combo it with 1 microbomb.

**Comment by Iamgoofball** [comment]
@Lumipharon Keep in mind dead mobs with damage are easier to gib than a 100% healthy player. A guy who's dead and then has these activate will probably gib. The 4 cap just guarantees you'll gib.

I think the bomb core thing sounds like a valid gameplay tactic and I don't see any problem with it.

**Comment by Lumipharon** [comment]
By the time the op is dead he'll be butt naked and it'll be too late.

Being able to literally blow your enemies apart with a macrobomb is cool, but gear denial is important as fuck - ever since ops lost their default implants, every op round where they don't just blitz the station, the crew always ens up with piles of loot.

This wont help in that regard, since it's still a case of spending limited resources planning to die rather then on surviving and winning in the first place.

**Comment by PKPenguin321** [comment]
i will +1 if and only if you leave behind the implant that just gibs you with little to no explosion so that we can some day give them back to ops at roundstart

**Comment by Iamgoofball** [comment]
@Lumipharon *deathgasp

**Comment by Lumipharon** [comment]
I was specifically talking about how you said 'dead bodies with damage gib easier'.
Deathgasping works when you're alive, so is a moot point.

Also, my point about gear denial aside, wouldn't it make more sense to have larger increments? Like 2 or 4tc microbombs instead of 1?

**Comment by Iamgoofball** [comment]
@Lumipharon One of the things about this being 1tc is that these are available in normal Traitor rounds too.

So if you've bought all the gear you really want, and have like 3 TC left or something, and don't really want anything else, buy a couple implanters and pop 'em in incase you die, as a final "fuck you" to whoever killed you.

**Comment by palpatine213** [comment]
@Lumipharon Nah, you gotta have 1 TC increments for maximum autism custom sized explosion.

**Comment by Kromgar** [comment]
For fucks sake. Make Gib yourselftier implants cost 1tc. Atleast then its about the price of an emag.

At this point it is still 20 tc for making sure your entire team won't lose their gear. If someone refuses then sure thats only 16 tc... Thats a shitload of TC.

**Comment by Anonus** [comment]
If a traitor with the "die gloriously" objective can get near their target they can explosively murder them, which, while glorious, is kind of bullshit.

**Comment by kevinz000** [comment]
They could do it with toxins
And doesn't even take TC

**Comment by Iamgoofball** [comment]
@Anonus There's no problem with that because they can already do it for free with toxins.

**Comment by Lumipharon** [comment]
Almost everything a traitor can buy in the uplink can be substituted by something that can get elsewhere, so it's not a very good argument.

**Comment by Iamgoofball** [comment]
Then there's no problem here.

**Comment by Anonus** [comment]
"For free with toxins" as an assistant, bartender, chef, botanist, librarian, chaplain, lawyer, engineer, cargo tech, medical doctor, geneticist, or virologist.
Right. The chances of any of those being able to get a bomb from toxins is really, really low.

**Comment by Iamgoofball** [comment]
Okay.

File: code/game/objects/items/weapons/implants/implant.dm [hunk 1]
```diff
@@ -101,12 +101,37 @@
...
 		return 0
-	explosion(src,0,1,5,7,10, flame_range = 5)
-	if(imp_in)
+	var/light = 2
+	var/medium = 1
+	var/heavy = 0.5
+	for(var/obj/item/weapon/implant/explosive/E in imp_in)
+		heavy += 0.5
+		medium += 1
+		light += 2
```
Comment (Anonus) [inline code/game/objects/items/weapons/implants/implant.dm]: Wait, why would you buy the macrobomb implant at all? You hit maxcap with 10 of these, all that increases past that is flame range and that's not really worth it.

Comment (Iamgoofball) [inline code/game/objects/items/weapons/implants/implant.dm]: I believe I fucked up my math, I'll mess with the macrobomb then.

**Comment by WJohn** [comment]
Yeah personally I'd like it if you only needed 1 microimplant to gib, with a tiny itty bitty explosion so all nuke ops can cheaply buy equipment denial implants.

**Comment by Fox-McCloud** [comment]
Ok, for microbomb mechanics:

- only takes a single microbomb to gib you
- Macrobombs are not the full size of a max-size explosion; it's more like 80-90%


IMO, Nuke ops should start out *pre-injected* with these, but that's a separate PR.

**Comment by Iamgoofball** [comment]
Alrighty, I'll make the adjustments and cheapen macrobombs.

**Comment by Iamgoofball** [comment]
Nuke ops start with microbombs again.

**Comment by Cheridan** [comment]
:+1:

**Comment by Lumipharon** [comment]
do they start with enough to gib? Otherwise it's kinda pointless.

**Comment by Fox-McCloud** [comment]
Ahhh yeah!

big :+1:  from me now that they gib whoever they're implanted in no matter how many!

**Comment by Iamgoofball** [comment]
@Lumipharon All it takes is 1 to gib now.

File: code/game/objects/items/weapons/implants/implant.dm [hunk 2]
```diff
@@ -101,12 +101,36 @@
...
+		heavy += 0.25
+		medium += 0.5
+		light += 1
+		if(E != src)
+			qdel(E)
+	heavy = round(heavy)
+	medium = round(medium)
+	light = round(light)
+	imp_in.gib()
+	explosion(src,0,heavy,medium,light,light, flame_range = medium)
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/implants/implant.dm]: would recommend tying flame range to light as opposed to medium--generally flame range is typically identical to the light range.

File: code/game/objects/items/weapons/implants/implant.dm [hunk 3]
```diff
@@ -101,12 +101,36 @@
...
+	name = "macrobomb implant"
+	desc = "And boom goes the weasel. And everything else nearby."
+	icon_state = "explosive"
 
+/obj/item/weapon/implant/explosive/macro/activate(var/cause)
+	if(!cause || !imp_in)	return 0
+	if(cause == "action_button" && alert(imp_in, "Are you sure you want to activate your macrobomb implant? This will cause you to explode and gib!", "Macrobomb Implant Confirmation", "Yes", "No") != "Yes")
+		return 0
+	imp_in.gib()
+	explosion(src,0,5,10,20,20, flame_range = 5)
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/implants/implant.dm]: Same here, and if that's too much, at least up this to 10; 5 isn't going to impact anyone as they're gibbed or dead at that stage.

**Comment by Fox-McCloud** [comment]
It just occurred to me; your arguments for explosions are incorrect.

explosion(src,0,5,10,20,20) will result in a a size 0,5,10,20 (devastation, heavy, light, and flash) and adminlog set to 20. Uhhhh.

**Comment by Alek2ander** [comment]
Speaking of adminlog, please add logging + admin alerts.

**Comment by Iamgoofball** [comment]
You already get them. The explosion logs.

**Comment by Fox-McCloud** [comment]
@Alek2ander 

Admin logging for explosions is turned on by default; you specifically have to override it to turn it off (and very few do).

**Comment by Alek2ander** [comment]
It's also kinda useful to know exactly what and/or who caused the explosion.

**Comment by optimumtact** [comment]
Fuck nigga, if they want logging give them more logging okay

**Comment by Steelpoint** [comment]
:+1: On Ops starting with explosive implants again.

**Comment by Iamgoofball** [comment]
@Steelpoint Both. Traitors can buy microbombs, and 20 microbombs = macrobomb, so why not basically.

**Comment by Kromgar** [comment]
Lets just wait for Cheridan to say no to this PR because it interferes with Roleplay.

He was vehemently opposed to nuke ops starting with implants

**Comment by Iamgoofball** [comment]
@Kromgar Maybe if you read the PR comments for once you'd see him :+1: ing it

**Comment by Kromgar** [comment]
All he did before was bitch and moan that explsoive implants ruined RP and he would not allow roundstart implants again. I am so confused

**Comment by Iamgoofball** [comment]
This is ready for merge.

**Pull request merged**